aptos-move/aptos-release-builder/src/validate.rs

```diff
@@ -0,0 +1,379 @@
+    proposal_id: U64,
+}
+
+impl NetworkConfig {
+    pub fn new_from_dir(endpoint: Url, test_dir: PathBuf) -> Result<Self> {
+        let root_key_path = test_dir.join("mint.key");
+        let private_identity_file = test_dir.join("0/private-identity.yaml");
+        let private_identity =
+            serde_yaml::from_slice::<PrivateIdentity>(&fs::read(private_identity_file)?)?;
+
+        Ok(Self {
+            endpoint,
+            root_key_path,
+            validator_account: private_identity.account_address,
+            validator_key: private_identity.account_private_key,
+        })
+    }
+
+    pub fn submit_and_execute_proposal(&self, script_path: Vec<PathBuf>) -> Result<()> {
+        for path in script_path {
+            self.mint_to_validator()?;
+            let proposal_id = self.create_governance_proposal(path.as_path(), false)?;
+            self.vote_proposal(proposal_id)?;
+            sleep(Duration::from_secs(40));
+            self.add_proposal_to_allow_list(proposal_id)?;
+            self.execute_proposal(proposal_id, path.as_path())?;
+        }
+        Ok(())
+    }
+
+    pub fn submit_and_execute_mutli_step_proposal(&self, script_path: Vec<PathBuf>) -> Result<()> {
+        let first_script = script_path.first().unwrap();
+        let proposal_id = self.create_governance_proposal(first_script.as_path(), true)?;
+        self.vote_proposal(proposal_id)?;
+        // Wait for the proposal to resolve.
+        sleep(Duration::from_secs(40));
+        for path in script_path {
+            self.add_proposal_to_allow_list(proposal_id)?;
+            self.execute_proposal(proposal_id, path.as_path())?;
+        }
+        Ok(())
+    }
+
+    pub fn create_governance_proposal(
+        &self,
+        script_path: &Path,
+        is_multi_step: bool,
+    ) -> Result<u64> {
+        let address_string = format!("{}", self.validator_account);
+        let privkey_string = format!("{}", hex::encode(self.validator_key.to_bytes()));
+        let mut root_path = std::path::Path::new(env!("CARGO_MANIFEST_DIR")).to_path_buf();
+        root_path.pop();
+        root_path.pop();
+
+        let framework_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("..")
+            .join("..")
+            .join("aptos-move")
+            .join("framework")
+            .join("aptos-framework");
+
+        let mut args = vec![
+            "run",
+            "--bin",
+            "aptos",
+            "--",
+            "governance",
+            "propose",
+            "--pool-address",
+            address_string.as_str(),
+            "--script-path",
+            script_path.to_str().unwrap(),
+            "--framework-local-dir",
+            framework_path.as_path().to_str().unwrap(),
+            "--metadata-url",
+            "https://raw.githubusercontent.com/aptos-labs/aptos-core/b4fb9acfc297327c43d030def2b59037c4376611/testsuite/smoke-test/src/upgrade_multi_step_test_metadata.txt",
```

Comment:
Can we move this txt into this repo?

Reply:
Can we first commit this PR and then add this file? We need a permanent commit for this file.

Reply:
Gotcha. That works too